Major health analyzer readability improvements

googobbug opened the pull request:
## About The Pull Request

Makes quite a few changes to make health analyzers much easier to read:
- Useless "minor X damage" text removed
- Non-verbose mode still displays overall numbers instead of removing the entire damage value table (who uses this?)
- Heart attack indicator is now bold and put inside the actual scan text
- Missing organs displayed in organ section
- Large amounts of redundant information removed from the advanced scanner seriously it was so bad
- Recommended treatment text removed from health scan wounds readout. Use the dedicated wound mode if you want to see it.
- Normal scanner can now detect missing eyes and ears, as it could already detect organ damage in them. Blindness/deafness detection are still advanced-scanner exclusive.

And the big one: chemical allergies are now displayed when performing a reagent scan!

## Why It's Good For The Game

Makes the health analyzer slightly less impenetrable, makes the advanced scanner way less of an eyesore, puts important information in visible spots so people can actually see missing organs, heart attacks, and allergies. 

Seriously, just look at it!

![image](https://user-images.githubusercontent.com/19880843/143656859-80b72f61-fd7c-43ca-be9f-3f64c3572e02.png)
![image](https://user-images.githubusercontent.com/19880843/143657550-e8dcc199-cd96-4ffe-86e2-d95d8c780521.png)
![image](https://user-images.githubusercontent.com/19880843/143657629-dfa11297-4f42-401c-b733-07465a8882fb.png)


## Changelog

:cl: Bumtickley00
qol: Health analyzers have had significant readability improvements!
qol: Health analyzers display allergy information when performing a reagent scan.
/:cl:

Commit:
scanner

Commit:
de-shitcode

newfren commented:
the missing organ section is great

GoldenAlpharex commented:
That's really cool.

FernandoJ8 commented:
GOD, THANK YOU.
It's hard to overstate how big of a pain it is to be unable to tell if an organ is missing or not. This is awesome.

Farquaar commented:
Yikes, what did you do to my boy Cesar

AdmiralPancakes1 commented:
I assure you he deserved it

exymian commented:
How do we scan for body temp&blood level ? Can you detail out what kind of information you removed that you feel is useless ?

Don't get me wrong this should be done and is a good idea

newfren commented:
body temp/blood level is on the images shown

AdmiralPancakes1 commented:
the only info I removed was stuff that was already displayed elsewhere in the same scan

exymian commented:
thats my bad i thought the images were before/after type so i was confused

tralezab commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -264,58 +271,20 @@ GENE SCANNER
-								<td><font color='#cc3333'>[(org.brute_dam > 0) ? "[CEILING(org.brute_dam,1)]" : "0"]</font></td>\
-								<td><font color='#ff9933'>[(org.burn_dam > 0) ? "[CEILING(org.burn_dam,1)]" : "0"]</font></td></tr>"
+			if(mode == SCANNER_VERBOSE)
+				for(var/o in damaged)
+					var/obj/item/bodypart/org = o //head, left arm, right arm, etc.
```
**NOTE:** this suggestion changes the var name, you'll need to change other uses of `org` > `limb`. org is not a good var name!

```suggestion
				for(var/obj/item/bodypart/limb as anything in damaged)
```

tralezab commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -165,17 +165,17 @@ GENE SCANNER
 
+	render_list += "[span_info("Analyzing results for [M]:")]\n<span class='info ml-1'>Overall status: [mob_status]</span>\n"
+	
+	SEND_SIGNAL(M, COMSIG_LIVING_HEALTHSCAN, render_list, advanced, user, mode)
```
while you're here, go to the top of this proc and rename the var `M`, it's not a descriptive var

tralezab commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -184,36 +184,18 @@ GENE SCANNER
-			render_list += "<span class='alert ml-1'>Subject lacks a stomach.</span>\n"
 
 	if(iscarbon(M))
 		var/mob/living/carbon/C = M
```
make this var more descriptive, please

tralezab commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -237,17 +219,42 @@ GENE SCANNER
+		
+	//Eyes and ears
+	if(advanced && iscarbon(M))
+		var/mob/living/carbon/C = M
```
make this var more descriptive, please

tralezab commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -237,17 +219,42 @@ GENE SCANNER
 	// Body part damage report
-	if(iscarbon(M) && mode == SCANNER_VERBOSE)
+	if(iscarbon(M))
 		var/mob/living/carbon/C = M
```
make this var more descriptive, please

tralezab commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -264,58 +271,20 @@ GENE SCANNER
-		render_list += "<span class='info ml-1'>Eye status:</span>[message == "" ? "\n<span class='info ml-2'>Healthy.</span>" : message]\n"
-
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
```
make this var more descriptive, please

tralezab commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -458,9 +451,16 @@ GENE SCANNER
 			render_list += "<span class='notice ml-1'>Subject is temporally unstable. Stabilising agent is recommended to reduce disturbances.</span>\n"
-
+		
+		for(var/datum/quirk/quirky in M.quirks)
```
```suggestion
		for(var/datum/quirk/quirky as anything in M.quirks)
```

tralezab reviewed (changes requested):
some var cleanup while you're here, and use of `as anything`

Commit:
single letter var exorcism

googobbug committed:
Merge branch 'master' into healthy

AdmiralPancakes1 commented:
Alright, should be good to go.
Additional changes aside from style fixes:
Made the wounds readout on the health even more compact
![image](https://user-images.githubusercontent.com/19880843/145349208-36326724-b8bb-45f5-9d26-5877ecfce588.png)
Fixed some weird indent stuff that possibly made cybernetics not show up when scanning plasmamen (?)
"No addictions" text removed from chem scan, as the lack of big red text should be enough of an indicator

LemonInTheDark committed:
Sanitizes diseases, god they're messy (#63319)

tgstation-server committed:
Automatic changelog generation for PR #63319 [ci skip]

stylemistake committed:
Require Node 16, Disable experimental Yarn PnP ESM loader (#63314)

    Fixes problems with building tgui on certain versions of Node.
    Bumps required NodeJS version to 16 (which is the current LTS)

These two things are atomic, they're just bundled in one PR for convenience.

TheBonded committed:
Fixes buckling onto tiles with dense atoms (#63277)

Before this change, you could put a dense object (like a computer frame) over a chair and buckle to it. This fixes that

tgstation-server committed:
Automatic changelog generation for PR #63277 [ci skip]

SuperNovaa41 committed:
Fixes AI shells being instantly converted with the Twisted Construction spell (#63231)

* fixes that shit

* should work

Co-authored-by: SuperNovaa41 <supernovaa41@protonmail.com>

tgstation-server committed:
Automatic changelog generation for PR #63231 [ci skip]

AnturK committed:
Paintings improvements. (#63170)

About The Pull Request

    Paintings can now do stroke painting.
    Added painting management panel for admins.
    Paintings now display author's character name, year of painting, medium and patron when hung on wall.
    You can become new patron by paying more than the previous one.
    Added painter's palettes to library vendor. (Sprites by @Mickyan )

Backend changes:

    Images are now stored in /data/paintings/images/*.png instead of /data/paintings/[category]/*.png
    Old categories are now just tags

Screens & Video
Changelog

cl
add: You can now become patron of your favorite painting by buying sponsorship from Nanotrasen Trust Foundation.
add: Painter's palettes are now available at library vendor.
qol: Can use strokes in paintings now
/cl

tgstation-server committed:
Automatic changelog generation for PR #63170 [ci skip]

Fikou committed:
the deep fryer reacts to cold (#63073)

dunking a colder than 0 celsius person in the deep fryer causes additional damage
dunking a VERY cold person causes fun stuff
putting a freeze cube in the deep fryer causes !FUN! stuff

tgstation-server committed:
Automatic changelog generation for PR #63073 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

RandomGamer123 committed:
Adds documentation for and makes station_was_nuked flag only set if the station was actually nuked (#63107)

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #63107 [ci skip]

E231kei500bandai committed:
Increases volume of the sound effect for knocking on firelocks (#63053)

Changed firelocks to have the same bash and knock sound effects as windows.
Which increases the volume of the knock sound effect and also adds the bash sound effect.

Also improves some of the code for firelock and window

Firelocks already had the knock sound effect, but it is so quiet that you can barely hear it. The volume should be increased so that people can hear it.

RandomGamer123 committed:
Stops drones from using keycard authenticators (#63135)

See title. Drones were able to use these due to their internal id card having captain access

tgstation-server committed:
Automatic changelog generation for PR #63135 [ci skip]

EOBGames committed:
Moth Food Returns: A Taste of the Fleet (#63174)

tgstation-server committed:
Automatic changelog generation for PR #63174 [ci skip]

Seris02 committed:
fixes bots getting stuck on railings and crashing clients (#63306)

Co-authored-by: John Willard <53777086+JohnFulpWillard@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #63306 [ci skip]

tf-4 committed:
Checks for candidates in dynamic roundstart (#63305)

tgstation-server committed:
Automatic changelog generation for PR #63305 [ci skip]

JohnFulpWillard committed:
prevents closing out of the tgui tab from blowing you up (#63275)

tgstation-server committed:
Automatic changelog generation for PR #63275 [ci skip]

SuperNovaa41 committed:
fix (#63309)

Co-authored-by: SuperNovaa41 <supernovaa41@protonmail.com>

tgstation-server committed:
Automatic changelog generation for PR #63309 [ci skip]

CandleJaxx committed:
Fixes a potential oversight in regards to Runechat layer placement. (#63310)

tgstation-server committed:
Automatic changelog generation for PR #63310 [ci skip]

LemonInTheDark committed:
Fixes random hell call-counts to Move (#63317)

Removes the parent call from mob/Login

Mothblocks pinged me with a profile of just a hellish amount of move calls
Way too many, like 200000
Started looking into it, got distracted by how expensive macros were
Turns out most of the cost of macros was in the nuke ops summon spawner

So I looked at why, only bit of it that was at all expensive was the login for the cyborgs
Tested on a local, huh that is slow yeah

Looked at the profiler, huh there's that move count again
So anyway, why is login so expensive

Spawned a few cyborgs in, dragged myself into them, nothing
Spawned myself in with sdql magic using the summon spawner, man it really is still there
I guess it has to do with move somehow, try and stick a breakpoint on it, get fucked by the debugger
I notice the hanging comes during the Login parent call

Try again, this time with good breakpoints.
We're trying to move, to (1,1,1), just like the reference says we will https://secure.byond.com/docs/ref/#/mob/proc/Login

But man, it just keeps happening, and we don't actually move
Step through the code, we've got that null loc check in atom/movable/Move

So the move to (1,1,1) isn't working

Here's the exact line from the reference
"If the mob has no location, place it near (1,1,1) if possible"
Keyword is near

Talked to lummox about this behavior, figured it was a bug

It turns out by near, they mean inside that tile's area
It'll keep trying to place you somewhere, in an attempt to effectively cover for shitty login systems, until you succeed in moving

That tile is space. There are 200,000 tiles with the same area as it
OH.

So anyway, we're not calling parent on mob/login anymore. We can do all the work it did that we care about ourselves (IE: Just the statobj set)
And this way we don't need to worry about 4 SECONDS OF OVERTIME WHENEVER SOME POOR FUCK MESSES UP SPAWN ORDER

So yeah, I'm a genius and not at all just malding at the existance of keybind macros, and hopefully another source of stutter bites the dust
Not actually sure how widespread this is, but even if it's just spawn becacons that's pretty banging

tgstation-server committed:
Automatic changelog generation for PR #63317 [ci skip]

SuperNovaa41 committed:
F2 works for quick screenshotting again (#63293)


Co-authored-by: SuperNovaa41 <supernovaa41@protonmail.com>

tgstation-server committed:
Automatic changelog generation for PR #63293 [ci skip]

MrMelbert committed:
Stops Starlight Condensation from runtiming to hell and back (+ cleans up the code a bit)  (#63300)

tgstation-server committed:
Automatic changelog generation for PR #63300 [ci skip]

jlsnow301 committed:
TGUI list input conversions (#63315)

Co-authored-by: tralezab <40974010+tralezab@users.noreply.github.com>
Co-authored-by: Sealed101 <75863639+Sealed101@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #63315 [ci skip]

Jackraxxus committed:
Fixes deconning an AI shell ghosting the AI (#63247)

tgstation-server committed:
Automatic changelog generation for PR #63247 [ci skip]

carshalash committed:
Lowers reagent size of spider bites.  (#63117)

tgstation-server committed:
Automatic changelog generation for PR #63117 [ci skip]

Mothblocks commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -264,147 +271,136 @@ GENE SCANNER
 			for(var/k in wounded_part.wounds)
 				var/datum/wound/W = k
-				render_list += "<div class='ml-2'>Type: [W.name]\nSeverity: [W.severity_text()]\nRecommended Treatment: [W.treat_text]</div>\n" // less lines than in woundscan() so we don't overload people trying to get basic med info
+				render_list += "<div class='ml-2'>[W.name] ([W.severity_text()])</div>" // less lines than in woundscan() so we don't overload people trying to get basic med info
```
Bring back the `treat_text`, wound scans honestly don't need to exist and I'd like to remove them someday. The venn diagram of people who know about wound scan mode and don't already know how to solve wounds is very minimal, and switching between them is clunky.

Mothblocks commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -264,147 +271,136 @@ GENE SCANNER
 		for(var/i in wounded_parts)
 			var/obj/item/bodypart/wounded_part = i
-			render_list += "<span class='alert ml-1'><b>Warning: Physical trauma[LAZYLEN(wounded_part.wounds) > 1? "s" : ""] detected in [wounded_part.name]</b>"
+			render_list += "<span class='alert ml-1'><b>Physical trauma[LAZYLEN(wounded_part.wounds) > 1? "s" : ""] detected in [wounded_part.name]</b>"
```
```suggestion
			render_list += "<span class='alert ml-1'><b>Physical trauma[LAZYLEN(wounded_part.wounds) > 1 ? "s" : ""] detected in [wounded_part.name]</b>"
```

Mothblocks commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -264,147 +271,136 @@ GENE SCANNER
+			if(mode == SCANNER_VERBOSE)
+				for(var/obj/item/bodypart/limb as anything in damaged)
+					dmgreport += "<tr><td><font color='#cc3333'>[capitalize(limb.name)]:</font></td>\
+									<td><font color='#cc3333'>[(limb.brute_dam > 0) ? "[CEILING(limb.brute_dam,1)]" : "0"]</font></td>\
```
Don't use tab alignment, this is banned in the style guide.
Split this into three `dmgreport += ""` instead.

Mothblocks commented:
> Non-verbose mode still displays overall numbers instead of removing the entire damage value table (who uses this?)

No one, and like all verbs it should be axed.

Commit:
verb exorcism + style cleanup

Commit:
help me

AdmiralPancakes1 commented:
wow yeah idk what I did to this branch but all my maps are conflicting and now it can't find the python executable

will fix after finals probably

Commit:
Merge branch 'healthy' of https://github.com/AdmiralPancakes1/tgstation into healthy

Commit:
please god help me

AdmiralPancakes1 commented:
oh god damn it

AdmiralPancakes1 commented:
reopening on new branch
